### PR TITLE
React/smaa results

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

**Review by kzalite — APPROVED** [review]

File: app/ts/SmaaTab/SmaaResults/SmaaWeightsTable/SmaaWeightsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            <TableRow>
+              {_.map(filteredCriteria, (criterion: ICriterion) => (
+                <TableCell key={criterion.id}>{criterion.title}</TableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            <TableRow>
+              {_.map(
+                filteredCriteria,
```
Comment (DanielReid) [inline app/ts/SmaaTab/SmaaResults/SmaaWeightsTable/SmaaWeightsTable.tsx:46]: extract?

File: app/ts/SmaaTab/SmaaTab.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+              <PreferencesContextProviderComponent
+                scenarios={scenarios}
+                currentScenarioId={currentScenarioId}
+                workspaceId={workspaceId}
+                problem={problem}
+                updateAngularScenario={updateAngularScenario}
+              >
+                <SmaaResultsContextProviderComponent>
+                  <ErrorHandler>
+                    <SmaaResults />
```
Comment (DanielReid) [inline app/ts/SmaaTab/SmaaTab.tsx:69]: nice staircase. I wonder if this is an antipattern. Seems OK

**Review by DanielReid — APPROVED** [review]
very nice

File: app/ts/SmaaTab/SmaaResults/CentralWeights/CentralWeightsPlot/CentralWeightsPlot.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    centralWeights,
+    filteredCriteria,
+    filteredAlternatives,
+    undefined
+  );
+  generate(settings);
+
+  return (
+    <Grid item>
+      <div style={{width: width, height: height}} id={'central-weights-plot'} />
```
[the remark this answers is not in the document]
Comment (chinook25) [reply app/ts/SmaaTab/SmaaResults/CentralWeights/CentralWeightsPlot/CentralWeightsPlot.tsx:26]: Yep

File: app/ts/SmaaTab/SmaaResults/SmaaResults.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  const {smaaWeights, ranks, centralWeights} = useContext(SmaaResultsContext);
+
+  return (
+    <Grid container spacing={2}>
+      <ScenarioSelection
+        scenarios={scenarios}
+        currentScenario={currentScenario}
+      />
+      <UncertaintyOptions />
+      <Grid item xs={12}>
```
[the remark this answers is not in the document]
Comment (chinook25) [reply app/ts/SmaaTab/SmaaResults/SmaaResults.tsx:24]: Tried adding them to Karlis' code. Stuff started breaking when I did, didn't have the energy to investigate, so left it like it is.

File: app/ts/SmaaTab/SmaaResults/SmaaResultsUtil.ts [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+import {ChartConfiguration, Primitive} from 'c3';
+import {format} from 'd3';
+import _ from 'lodash';
+
+export function hasStochasticMeasurements(
+  filteredDistributions: Distribution[]
+): boolean {
+  return _.some(
+    filteredDistributions,
+    (distribution: Distribution) =>
```
[the remark this answers is not in the document]
Comment (chinook25) [reply app/ts/SmaaTab/SmaaResults/SmaaResultsUtil.ts:30]: I don't think we've ever implemented that. Shall check trello.

[the remark this answers is not in the document]
Comment (chinook25) [reply app/ts/SmaaTab/SmaaResults/SmaaResultsUtil.ts:30]: I can't find it.

**Review by DanielReid — APPROVED** [review]
good

**Pull request merged**